### PR TITLE
[source-kafka] Fix subcommand's long, short, Use values and update docs

### LINKED ISSUE #59: [source-kafka] Fix subcommand's long, short, Use values and update docs
see https://github.com/knative/client-contrib/pull/33#pullrequestreview-438048267

### PR DESCRIPTION
**Pull request by daisy-ycguo**
Closes #59

### CODE REVIEW

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 1]
```diff
@@ -55,22 +55,23 @@ func (f *kafkaSourceCommandFactory) SourceCommand() *cobra.Command {
...
 	sourceCmd.Short = "Knative eventing Kafka source plugin"
-	sourceCmd.Long = "Manage your Knative Kafka eventing sources"
+	sourceCmd.Long = "Manage Knative Kafka eventing sources"
 	return sourceCmd
 }
 
 func (f *kafkaSourceCommandFactory) CreateCommand() *cobra.Command {
 	createCmd := f.defaultCommandFactory.CreateCommand()
-	createCmd.Short = "create NAME"
+	createCmd.Use = "create NAME"
```
Comment (navidshaikh) [inline plugins/source-kafka/pkg/factories/command_factory.go:64]: this should represent all the flags which are mandatory for creating a kafka source

(check ping source usage string for example where `--sink` is a mandatory option)

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 2]
```diff
@@ -82,8 +83,8 @@ func (f *kafkaSourceCommandFactory) UpdateCommand() *cobra.Command {
 
 func (f *kafkaSourceCommandFactory) DescribeCommand() *cobra.Command {
 	describeCmd := f.defaultCommandFactory.DescribeCommand()
-	describeCmd.Short = "describe NAME"
-	describeCmd.Long = "update a Kafka source"
+	describeCmd.Use = "describe NAME"
+	describeCmd.Short = "update a Kafka source"
```
Comment (navidshaikh) [inline plugins/source-kafka/pkg/factories/command_factory.go:87]:
```suggestion
	describeCmd.Short = "Describe a Kafka source"
```

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 3]
```diff
@@ -55,22 +55,23 @@ func (f *kafkaSourceCommandFactory) SourceCommand() *cobra.Command {
...
 kn source kafka create mykafkasrc --servers my-cluster-kafka-bootstrap.kafka.svc:9092 --topics test-topic --consumergroup test-consumer-group --sink svc:event-display`
 	return createCmd
 }
 
 func (f *kafkaSourceCommandFactory) DeleteCommand() *cobra.Command {
 	deleteCmd := f.defaultCommandFactory.DeleteCommand()
-	deleteCmd.Short = "delete NAME"
-	deleteCmd.Long = "delete a Kafka source"
+	deleteCmd.Use = "delete NAME"
+	deleteCmd.Short = "delete a Kafka source"
```
Comment (navidshaikh) [inline plugins/source-kafka/pkg/factories/command_factory.go:74]:
```suggestion
	deleteCmd.Short = "Delete a kafka source"
```

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 4]
```diff
@@ -55,22 +55,23 @@ func (f *kafkaSourceCommandFactory) SourceCommand() *cobra.Command {
...
-	sourceCmd.Long = "Manage your Knative Kafka eventing sources"
+	sourceCmd.Long = "Manage Knative Kafka eventing sources"
 	return sourceCmd
 }
 
 func (f *kafkaSourceCommandFactory) CreateCommand() *cobra.Command {
 	createCmd := f.defaultCommandFactory.CreateCommand()
-	createCmd.Short = "create NAME"
+	createCmd.Use = "create NAME"
+	createCmd.Short = "create a kafka source"
```
Comment (navidshaikh) [inline plugins/source-kafka/pkg/factories/command_factory.go:65]:
```suggestion
	createCmd.Short = "Create a kafka source"
```

File: plugins/source-kafka/pkg/factories/command_factory.go (hunk 1 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go:64]: fixed.

File: plugins/source-kafka/pkg/factories/command_factory.go (hunk 2 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go:87]: fixed.

File: plugins/source-kafka/pkg/factories/command_factory.go (hunk 3 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go:74]: Fixed.

File: plugins/source-kafka/pkg/factories/command_factory.go (hunk 4 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go:65]: Fixed.

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 5]
```diff
@@ -55,22 +55,23 @@ func (f *kafkaSourceCommandFactory) SourceCommand() *cobra.Command {
...
+	sourceCmd.Long = "Manage Knative Kafka eventing sources"
 	return sourceCmd
 }
 
 func (f *kafkaSourceCommandFactory) CreateCommand() *cobra.Command {
 	createCmd := f.defaultCommandFactory.CreateCommand()
-	createCmd.Short = "create NAME"
+	createCmd.Use = "create NAME --sink SINK"
+	createCmd.Short = "Create a kafka source"
 	createCmd.Example = `#Creates a new Kafka source named as 'mykafkasrc' which subscribes a Kafka server 'my-cluster-kafka-bootstrap.kafka.svc:9092' at topic 'test-topic' using the consumer group ID 'test-consumer-group' and sends the event messages to service 'event-display'
```
Comment (chaozbj) [inline plugins/source-kafka/pkg/factories/command_factory.go:66]: Should we add a space between ```#``` and example description? I just found ```kn``` command has space between them.

Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go]: Fixed. Thank you.

File: plugins/source-kafka/README.md [hunk 6]
```diff
@@ -53,26 +53,26 @@ Manage your Knative Kafka eventing sources
 
 #### SEE ALSO
 
-* [kafka create](#kafka-create)	 - create NAME
-* [kafka delete](#kafka-delete)	 - delete NAME
-* [kafka describe](#kafka-describe)	 - describe NAME
+* [kafka create](#kafka-create)	 - Create a kafka source
+* [kafka delete](#kafka-delete)	 - Delete a Kafka source
+* [kafka describe](#kafka-describe)	 - Describe a Kafka source
```
Comment (navidshaikh) [inline plugins/source-kafka/README.md:58]: inconsistent spell for "kafka", should begin with lower case

File: plugins/source-kafka/README.md [hunk 7]
```diff
@@ -53,26 +53,26 @@ Manage your Knative Kafka eventing sources
...
+Create a kafka source
 
 #### Synopsis
 
-create NAME
+Create a kafka source
 
 ```
-kafka create NAME [flags]
+kafka create NAME --sink SINK [flags]
```
Comment (navidshaikh) [inline plugins/source-kafka/README.md:69]: I think all the flags except sink are required for kafka source creation.

Comment (navidshaikh) [reply plugins/source-kafka/README.md:69]: also, usage should also include being with `kn source`, i.e. 
```
kn source kafka create NAME --servers SERVERS --topics TOPICS --consumergroup GROUP [options]
```

File: plugins/source-kafka/README.md [hunk 8]
```diff
@@ -53,26 +53,26 @@ Manage your Knative Kafka eventing sources
...
 ```
 
 #### Examples
 
 ```
-#Creates a new Kafka source named as 'mykafkasrc' which subscribes a Kafka server 'my-cluster-kafka-bootstrap.kafka.svc:9092' at topic 'test-topic' using the consumer group ID 'test-consumer-group' and sends the event messages to service 'event-display'
+# Creates a new Kafka source named as 'mykafkasrc' which subscribes a Kafka server 'my-cluster-kafka-bootstrap.kafka.svc:9092' at topic 'test-topic' using the consumer group ID 'test-consumer-group' and sends the event messages to service 'event-display'
```
Comment (navidshaikh) [inline plugins/source-kafka/README.md:75]:
```suggestion
# Create a new Kafka source 'mykafkasrc' which subscribes a Kafka server 'my-cluster-kafka-bootstrap.kafka.svc:9092' at topic 'test-topic' using the consumer group ID 'test-consumer-group' and sends the events to service 'event-display'
```

File: plugins/source-kafka/pkg/factories/command_factory.go [hunk 9]
```diff
@@ -82,9 +83,9 @@ func (f *kafkaSourceCommandFactory) UpdateCommand() *cobra.Command {
 
 func (f *kafkaSourceCommandFactory) DescribeCommand() *cobra.Command {
 	describeCmd := f.defaultCommandFactory.DescribeCommand()
-	describeCmd.Short = "describe NAME"
-	describeCmd.Long = "update a Kafka source"
-	describeCmd.Example = `#Describes a Kafka source with NAME
+	describeCmd.Use = "describe NAME"
+	describeCmd.Short = "Describe a Kafka source"
+	describeCmd.Example = `# Describes a Kafka source with NAME
```
Comment (navidshaikh) [inline plugins/source-kafka/pkg/factories/command_factory.go:88]:
```suggestion
	describeCmd.Example = `# Describe a Kafka source with NAME
```

File: plugins/source-kafka/README.md (hunk 6 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/README.md:58]: fixed.

File: plugins/source-kafka/README.md (hunk 8 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/README.md:75]: fixed. Thank you.

File: plugins/source-kafka/pkg/factories/command_factory.go (hunk 9 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/pkg/factories/command_factory.go:88]: fixed.

File: plugins/source-kafka/README.md (hunk 7 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/README.md:69]: I have added all flags to the usage output.
As to add `kn source` to the usage, it's not easy to fix here. This part of README.md is auto generated by `cobra/doc`.  I could try to change my doc generator tool to add `kn source` in front of command usage. Let's keep it as it in this PR. I will create a new issue.

File: plugins/source-kafka/README.md [hunk 10]
```diff
@@ -53,26 +53,26 @@ Manage your Knative Kafka eventing sources
 
 #### SEE ALSO
 
-* [kafka create](#kafka-create)	 - create NAME
-* [kafka delete](#kafka-delete)	 - delete NAME
-* [kafka describe](#kafka-describe)	 - describe NAME
+* [kafka create](#kafka-create)	 - Create a kafka source
+* [kafka delete](#kafka-delete)	 - Delete a Kafka source
```
Comment (maximilien) [inline plugins/source-kafka/README.md:57]: Sorry but still seeing "Delete a Kafka source" which is not consistent with the "kafka" spelling. Am I missing something?

File: plugins/source-kafka/README.md [hunk 11]
```diff
@@ -94,11 +94,11 @@ kn source kafka create mykafkasrc --servers my-cluster-kafka-bootstrap.kafka.svc
 
 ### kafka delete
 
-delete NAME
+Delete a Kafka source
```
Comment (maximilien) [inline plugins/source-kafka/README.md:97]: Same here...

File: plugins/source-kafka/README.md (hunk 10 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/README.md]: change all command output to lower case, except the first word of a sentence.

File: plugins/source-kafka/README.md (hunk 11 above)
Comment (daisy-ycguo) [reply plugins/source-kafka/README.md]: change all command output to lower case, except the first word of a sentence.

**Pull request merged**